### PR TITLE
Home Task 3 + Home Task 4 + Home Task 5

### PR DESCRIPTION
**Pull request by yhlukhov**

### CODE REVIEW

File: src/components/Header/MovieDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { Logo } from '../common'
+import { MovieDescDiv, SearchIco } from './styled'
+import { formatRuntime } from '../../utilities'
+import { useRecoilState } from 'recoil'
+import { movieRecoil } from '../../recoilStore'
+
+const MovieDetails = () => {
+  const [movie, setMovie] = useRecoilState(movieRecoil)
+
+  return movie ? (
```
Comment (ValeriGolovina) [inline src/components/Header/MovieDetails.tsx:10]: you do not need to make a condition there cause you've done it earlier in src/components/Header/Top.tsx I think

File: src/components/body/MovieList.tsx [hunk 2]
```diff
@@ -1,21 +1,29 @@
...
+    // Getting movie list and filtering:
+    moviesList = genre === 'all' ? [...movies] : filter(movies, genre)
+  }
 
-export const MovieList: FC<PropsType> = ({ movies, sortBy, genre }) => {
-  // Getting movie list and filtering:
-  const moviesList = genre === 'all' ? [...movies] : filter(movies, genre)
   // Sorting:
-  sort(moviesList, sortBy)
+  sort(moviesList, sortParam)
```
Comment (ValeriGolovina) [inline src/components/body/MovieList.tsx:25]: Filtering, sorting is better to make in useEffect

File: src/components/Header/MovieDetails.tsx (hunk 1 above)
Comment (yhlukhov) [reply src/components/Header/MovieDetails.tsx:10]: Yes, if pass 'movie' via props (not use global state management)
i've changed code, now there is no condition
return (
    <MovieDescDiv>
      {movie.title}
      ....
    </MovieDescDiv>
  )

File: src/components/body/MovieList.tsx (hunk 2 above)
Comment (yhlukhov) [reply src/components/body/MovieList.tsx:25]: Agree. 
I've changed component. Now it uses "dispatch" inside useEffect function
export const MovieList = () => {
  const genre = useSelector(selectGenre)
  const sortBy = useSelector(selectSortBy)
  const searchTerm = useSelector(selectSearchTerm)
  const movies = useSelector(selectMovies)
  const page = useSelector(selectPage)
  const dispatch = useAppDispatch()

  useEffect(() => {
    dispatch(searchTerm ? searchMovies(searchTerm) : getMovies())
  }, [searchTerm, sortBy, genre, page, dispatch])

  return (
    <>
      <MovieCount />
      <MovieListDiv>
        {movies.map((movie) => (
          <MovieCard movie={movie} key={movie.id} />
        ))}
      </MovieListDiv>
      <Paginator />
    </>
  )
}

**Pull request merged**